### PR TITLE
JAMES-2524 Update of last libraries

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[aaf286e5d979c92da682500516dec69e4ae0f152] [ERROR] src/main/java/org/apache/james/metrics/dropwizard/DropWizardMetricFactory.java:[34] (imports) CustomImportOrder: Wrong lexicographical order for 'com.codahale.metrics.MetricRegistry' import. Should be before 'com.codahale.metrics.jmx.JmxReporter'.
```

checkstyle

File: server/container/guice/cassandra-guice/pom.xml [hunk 1]
```diff
@@ -259,11 +259,6 @@
             <groupId>org.awaitility</groupId>
             <artifactId>awaitility</artifactId>
         </dependency>
-        <dependency>
```
Comment (chibenwa) [inline server/container/guice/cassandra-guice/pom.xml]: Maybe @mbaechler  can explain a bit more about it, but it looks like something he added on purpose...

Comment (chibenwa) [reply server/container/guice/cassandra-guice/pom.xml]: IMO we should rather move this dependency to dropwizard metrics sub module anyway

File: pom.xml [hunk 2]
```diff
@@ -632,7 +632,7 @@
         <jackson.version>2.9.6</jackson.version>
         <jetty.version>9.4.7.v20170914</jetty.version>
         <testcontainers.version>1.8.1</testcontainers.version>
-        <metrics.version>3.2.6</metrics.version>
+        <metrics.version>4.0.3</metrics.version>
```
Comment (trantienduchn) [inline pom.xml]: upgrading dropwizard metrics leads to incompatible metrics version in cassandra driver dependency, see ci build error in https://github.com/linagora/james-project/pull/1560
Neither you declare old metric dependency go with cassandra driver dependency nor drop the metrics upgrade.

mvn dependency tree result:
```
[INFO] +- com.datastax.cassandra:cassandra-driver-core:jar:shaded:3.5.1:compile
[INFO] |  +- io.dropwizard.metrics:metrics-core:jar:3.2.6:compile
[INFO] |  +- com.github.jnr:jnr-ffi:jar:2.1.7:compile
```

Comment (chibenwa) [reply pom.xml]: Let's keep metrics at 3.2.6 IMO

File: server/container/guice/cassandra-guice/pom.xml (hunk 1 above)
Comment (chibenwa) [reply server/container/guice/cassandra-guice/pom.xml]: Well, that being said, dropwizard don't seem to be magically using the HdrHistogram back-end,
and something like https://bitbucket.org/marshallpierce/hdrhistogram-metrics-reservoir seems required...

I propose to keet @blackheaven  changes.

**Comment by chibenwa** [comment]
```
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR] Errors: 
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.defaultUnitShouldBeMilliseconds:198 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldAcceptEmptyThreshold:131 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldLoadGracePeriodInMs:188 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldReadXMLConfiguration:56 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldReadXMLConfigurationWhenRenderingInformationPartiallyOmited:94 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldThrowOnEmptyBodyTemplate:226 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldThrowOnEmptySubjectTemplate:207 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldThrowOnNegativeGracePeriod:178 » NoClassDefFound
[4ed8d0a1aaddaeb4d57bd83a8bafa60a6538e011] [ERROR]   QuotaMailingListenerConfigurationTest.fromShouldThrowOnNonParsableGracePeriod:168 » NoClassDefFound
```

**Comment by chibenwa** [comment]
I also reverted **commons-configuration** upgrade.

In **1.10.0** commons-collection, that we had hard time removing, is no more optional, leading to  `NoClassDefFoundError: org/apache/commons/collections/CollectionUtils`.

The long term solution is to move to commons-configuration2 anyway...

File: pom.xml [hunk 3]
```diff
@@ -2305,26 +2304,6 @@
                 <artifactId>bcprov-jdk15on</artifactId>
                 <version>1.60</version>
             </dependency>
-            <dependency>
```
Comment (mbaechler) [inline pom.xml]: by doing this, you don't drop jackson 1 (that is required by a several other modules) but you let the accidental resolution of deps happen. No doubt it's a bad idea

**Comment by chibenwa** [comment]
test this please (no build)

**Comment by chibenwa** [comment]
No more tested by CI

**Pull request closed without merging**